administrator/components/com_cronjobs/src/Controller/CronjobController.php

```diff
@@ -0,0 +1,163 @@
+
+use Exception;
+use Joomla\CMS\Application\AdministratorApplication;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\MVC\Controller\FormController;
+use Joomla\CMS\Router\Route;
+use Joomla\Component\Cronjobs\Administrator\Helper\CronjobsHelper;
+use function defined;
+
+/**
+ * The CronjobModel controller.
+ *
+ * @since __DEPLOY_VERSION__
+ */
+class CronjobController extends FormController
+{
+	/**
+	 * Add a new record
+	 *
+	 * @return boolean
+	 * @throws Exception
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function add(): bool
+	{
+		/**@var AdministratorApplication $app */
+		$app = $this->app;
+		$input = $app->getInput();
+		$validJobOptions = CronjobsHelper::getCronOptions();
+
+		$canAdd = parent::add();
+
+		if ($canAdd !== true)
+		{
+			return false;
+		}
+
+		$jobType = $input->get('type');
+		$jobOption = $validJobOptions->findOption($jobType) ?: null;
+
+		if (!$jobOption)
+		{
+			// ? : Is this the right redirect [review]
+			$redirectUrl = 'index.php?option=' . $this->option . '&view=select&layout=edit';
+			$this->setRedirect(Route::_($redirectUrl, false));
+			$app->enqueueMessage(Text::_('COM_CRONJOBS_ERROR_INVALID_JOB_TYPE'), 'warning');
+			$canAdd = false;
+		}
+
+		$app->setUserState('com_cronjobs.add.cronjob.cronjob_type', $jobType);
+		$app->setUserState('com_cronjobs.add.cronjob.cronjob_option', $jobOption);
+
+		// TODO : Parameter array handling below?
+
+		return $canAdd;
+	}
+
+	/**
+	 * Override parent cancel method to reset the add cronjob state
+	 *
+	 * @param   null  $key  Primary key from the URL param
+	 *
+	 * @return boolean  True if access level checks pass
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function cancel($key = null): bool
+	{
+		$result = parent::cancel($key);
+
+		$this->app->setUserState('com_cronjobs.add.cronjob.cronjob_type', null);
+		$this->app->setUserState('com_cronjobs.add.cronjob.cronjob_option', null);
+
+		// ? Do we need to redirect based on URL's 'return' param? {@see ModuleController}
+
+		return $result;
+	}
+
+	/**
+	 * Edit an existing record
+	 *
+	 * ! : Just acting as a proxy to the parent method atm
+	 *     Due removal if no additional handling needed here ⚠
+	 *
+	 * @param   string  $key     Name of primary key from urlVar
+	 * @param   string  $urlVar  Name of urlVar if different from primary key [?]
+	 *
+	 * @return boolean  True if access user has sufficient privileges
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function edit($key = null, $urlVar = null): bool
+	{
+		// TODO: Change the autogenerated stub
+		return parent::edit($key, $urlVar);
+	}
+
+	/**
+	 * Check if user has permissions to add an entry
+	 *
+	 * ! : Does nothing at the moment
+	 *     Remove if no special handling required ⚠
+	 *
+	 * @param   array  $data  Array of input data
+	 *
+	 * @return boolean  True if user can add an entry
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	protected function allowAdd($data = array()): bool
+	{
+		// TODO: Change or remove
```

Comment:
```suggestion
		// @todo Change or remove
```

The `@todo` tag is parsed by static analysing tools and IDE's better than just `TODO`

Reply:
Thank you for this suggestion!

Reply:
Resolved by 62deeb495cc69eb6f820751cbb816ed7bb8a7cd8. Thank you!